components/tidb_query_vec_expr/src/impl_math.rs

```diff
@@ -506,18 +508,18 @@ pub fn truncate_int_with_int(arg0: &Option<Int>, arg1: &Option<Int>) -> Result<O
 
 #[inline]
 #[rpn_fn]
-pub fn truncate_int_with_uint(arg0: &Option<Int>, arg1: &Option<Int>) -> Result<Option<Int>> {
-    match (arg0, arg1) {
-        (&Some(x), &Some(_)) => Ok(Some(x)),
+pub fn truncate_int_with_uint(arg0: Option<&Int>, arg1: Option<&Int>) -> Result<Option<Int>> {
+    match (arg0.cloned(), arg1.cloned()) {
```

Comment:
Maybe we need to avoid these clones, since which is not efficient.

Reply:
I'll leave this part of work on last stage of this project. If we `merge_null` before calling these vectorized functions, we do not even need to match these `Option`. Therefore, I prefer not to refactor these code for now :(

Reply:
Looks fine to me as long as there is no notable performance regression according to benchmarks (since there is still a long way until reaching the final implementation).

Reply:
I've changed from `Option<&usize>` to `usize` in latest commit